### ISSUE TITLE
Zettlr Cannot be Started in the Background

### ISSUE THREAD

**Issue by petervaro** [Zettlr/Zettlr#5415]
### Description

# TL;DR

`zettlr` cannot be started from an _application launcher_ (such as [`rofi`](https://github.com/davatorium/rofi)) or in fact, in any other ways that starts it _in the background_&mdash;worst case scenario, this freezes the entire X session without any indication as to why.

# Detailed Explanation

If `zettlr` is started as a foreground process, i.e.

```bash
$ zettlr
```

Then it happily runs the program after it printed out the following logs:

```text
[Application] Performing environment check ...
[Application] Pandoc has not been bundled with this release. Falling back to system version instead.
[Application] Quarto not found on system. *.qmd-files will be exported with Pandoc.
[Application] Environment check complete.
```

However, if it is started in the background, i.e.

```bash
$ zettlr &
```

It behaves very unexpectedly&mdash;it prints out the first log line:

```text
[Application] Performing environment check ...
```

And then it stops the process.  When brought back to the foreground with `fg`, it happily continues with printing out the rest of the logs which is then followed by opening the main window.

However, should the process remain in the background, when the current shell-session terminates, it produces the following two pop-ups, one after another:

![Screenshot_20241006_224711](https://github.com/user-attachments/assets/bf78fdc1-72c7-4445-896b-568567acd03c)

![Screenshot_20241006_224745](https://github.com/user-attachments/assets/916d3971-b6a3-45c2-a088-60e982add81c)

## Additional Information

> [!NOTE]
> The following information might be unrelated, but I'm sharing them nonetheless in case they are actually useful.

When started via `rofi` (via `drun` to be precise, i.e. the `.desktop` file's `Exec` entry is executed, which is just `zettlr`:

```ini
[Desktop Entry]
Version=1.0
Type=Application
Name=Zettlr
Comment=A Markdown Editor for the 21st century
Exec=zettlr
Icon=zettlr
Terminal=false
StartupNotify=false
Categories=Office;
MimeType=text/markdown;
StartupWMClass=Zettlr
```

) it freezes the entire X session.  When the X session is killed a nebulous error referring to `org.freedesktop.impl.portal.FileChooser` can be seen sometimes.

After digging deeper into this, it turned out that the `FileChooser` should be provided by `xdg-desktop-portal`.  However, my setup consists only of a Window Manager and no Desktop Environment, as such, it does not really provide anything, since no backend was installed.

But even after installing the GTK one and following the [suggestions described in this article](https://wiki.archlinux.org/title/XDG_Desktop_Portal#Configuration) of overriding the `XDG_CURRENT_DESKTOP` environment variable through a drop-in `systemd` snippet to force the GTK fallback and / or configuring the `xdg-desktop-portal` in the appropriate `portals.conf`, the behaviour did not change and the above described problem could still be reproduced.

### Reproducing

Start `zettlr` and fork it in the background:

```bash
> $ zettlr &
```

### Zettlr version

3.2.2

### Installation Method

Other (please specify below)

### Installation Method (Other)

Official Arch Repository (https://archlinux.org/packages/extra/x86_64/zettlr)

### Your Platform

- [ ] Windows
- [ ] macOS
- [X] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit (includes Apple Silicon)

### Operating System Version

Arch Linux (Kernel 6.11.2)

**Comment by nathanlesage** [comment]
There seem to be several components to this. You seem to have some specifics in what your OS is running, correct? In that you don't have a desktop environment? If so, could you please check if this behavior also happens on a "regular" live OS? If not, it is something with your computer. If you can reproduce this behavior, then it has something to do with the app.

I suspect that the binary doesn't like if you detach the stdout stream (which the errors you've been getting indicate) which is precisely what happens if you bonk it from the shell by forking it.

**Comment by petervaro** [comment]
> You seem to have some specifics in what your OS is running, correct?

Not having a DE is quite common among Arch users who wanted to have something very light and completely customised (or at least it used to be), but even if it is _strange_ and rather _specific_, all of the other Electron-based (or in fact, just Node-based) programs are running happily for the last decade or so on this installation (VSCodium, Signal, Beaver Notes, etc.), so I'm fairly certain, that the way Zettlr is distributed / packaged is the problem and not my setup.

> [...] could you please check if this behavior also happens on a "regular" live OS?

I am not exactly sure what you are asking me to do, I spent enough time debugging this already and if I'm being honest, I do not wish to go down another rabbit hole of replicating a similar setup on a live system, which by definition cannot be the same as it will have an entire DE.  (I quickly ran the latest (2024.10.01) Arch Linux with XFCE in a VM and I was able to start `zettlr` but only with the `--no-sandbox` flag as I was running the entire DE as `root`.  Does this help with anything, I wonder? :thinking: )

> I suspect that the binary doesn't like if you detach the stdout stream [...]

That was _kind of_ my immediate thought as well, however, that is incorrectly assumes a background process does not receive a `stdout` (and a `stderr`) which is of course not true, as demonstrated above, the background process inherits these from the forking session, that's how its logs could be observed.  But to put your mind at ease, if I explicitly provide files for these, the erroneous behaviour still remains:

```bash
$ zettlr > /tmp/stdout 2> /tmp/stderr &
```

**Comment by nathanlesage** [comment]
Hmhm, also it's a good indicator that other Electron apps work just fine in your setup …

@kyaso I don't have my Linux setup with me right now, so I can't dig into the issue, do you maybe have some bandwidth in the coming weeks to look at this?

**Comment by kyaso** [comment]
> @kyaso I don't have my Linux setup with me right now, so I can't dig into the issue, do you maybe have some bandwidth in the coming weeks to look at this?

Sure, I'll have it on my radar.

**Comment by Prateek-coder19** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by benniekiss** [comment]
@petervaro does this issue still occur for you on a recent version of Zettlr (4.6)?

**Comment by petervaro** [comment]
@benniekiss I stopped experimenting with Zettlr for the above reasons almost two years ago&mdash;it was unacceptable that the whole X session needed to be restarted if by accident I ran Zettlr in a way that I do with every other program.  Therefore, I have no idea whether this still occurs, but I provided enough information that it should be fairly simple to reproduce it in a VM.

**Comment by nathanlesage** [comment]
@benniekiss If you'd like to give it a shot, feel free to. I don't have the time to spin up a Linux VM right now. Also since there have been no other indications that others have a similar problem, despite the Arch package of Zettlr being quite popular, I don't think the issue persists or at least that it isn't an issue for most people.
